### DISCUSSION TITLE
Game doesn't obey window commands on multiple monitors

### DISCUSSION THREAD

**Discussion by spencer3035** [hyprwm/Hyprland#6294] [Q&A]
I'm having an issue with a steam game window not having the appropriate behavior after it opens.

When I launch the game (Trackmania), it open briefly on the monitor I am focused on (DP-1) and then immediately snaps to be on my other monitor (HDMI-A-1). It keeps the resolution of DP-1 and it thinks it is in fullscreen, but it is on a larger monitor so I can see through to the other applications on that workspace. 

I am further unable to modify the location or resolution of the window with any of the following commands: 

- `movewindow`
- `movetoworkspace`
- `movetoworkspacesilent`
- `fullscreen`
- `togglefloating` can have some effect, but it usually reverts after any other action

I don't have any application specific workspace rules. I do have rules that put odd workspaces on my first monitor and even workspaces on my second monitor, but these rules work fine for all other windows. 

I have further tried swapping the layout in multiple ways. All of these still result in the same behavior of the application snapping to the left side of HDMI-A-1. 

- Change `monitors` to have 0x0 for HDMI-A-1 and 1920x0 for DP-1. (origin at HDMI-A-1)
- Change `monitors` to have -2560x0 for HDMI-A-1 and 0x0 for DP-1. (origin at DP-1)
- Change `monitors` to have 1920x0 for HDMI-A-1 and 0x0 for DP-1. (Reversed, origin at DP-1)

It seems that for some reason the application always snaps to the HDMI-A-1 monitor because of the order it is plugged into on my graphics card, not due to any hyprland settings. 

Any help is appreaciated. I've been really frustrated with this issue. 

Here is the details about my monitors. Let me know if any other output would be helpful. 
```
$ hyprctl monitors all
Monitor HDMI-A-1 (ID 0):
	2560x1080@59.97800 at 1920x0
	description: LG Electronics LG ULTRAWIDE 0x00039818
	make: LG Electronics
	model: LG ULTRAWIDE
	serial: 0x00039818
	active workspace: 1 (1)
	special workspace: 0 ()
	reserved: 0 35 0 0
	scale: 1.00
	transform: 0
	focused: yes
	dpmsStatus: 1
	vrr: 0
	activelyTearing: false
	disabled: false
	currentFormat: XRGB8888
	availableModes: 2560x1080@59.98Hz 2560x1080@74.99Hz 1920x1080@75.00Hz 1920x1080@59.99Hz 1920x1080@59.94Hz 1920x1080@50.00Hz 1680x1050@59.95Hz 1600x900@60.00Hz 1280x1024@75.03Hz 1280x1024@60.02Hz 1280x800@59.81Hz 1152x864@60.00Hz 1280x720@60.00Hz 1280x720@59.94Hz 1280x720@50.00Hz 1024x768@75.03Hz 1024x768@60.00Hz 800x600@75.00Hz 800x600@60.32Hz 720x576@50.00Hz 720x480@59.94Hz 640x480@75.00Hz 640x480@59.94Hz 640x480@59.93Hz

Monitor DP-1 (ID 1):
	1920x1080@144.00101 at 0x0
	description: AOC 2590G4 0x0000593B
	make: AOC
	model: 2590G4
	serial: 0x0000593B
	active workspace: 2 (2)
	special workspace: 0 ()
	reserved: 0 35 0 0
	scale: 1.00
	transform: 0
	focused: no
	dpmsStatus: 1
	vrr: 0
	activelyTearing: false
	disabled: false
	currentFormat: XRGB8888
	availableModes: 1920x1080@60.00Hz 1920x1080@144.00Hz 1920x1080@119.98Hz 1920x1080@99.93Hz 1920x1080@59.94Hz 1920x1080@50.00Hz 1680x1050@59.88Hz 1280x1024@75.03Hz 1280x1024@60.02Hz 1440x900@59.90Hz 1280x720@60.00Hz 1280x720@59.94Hz 1280x720@50.00Hz 1024x768@119.99Hz 1024x768@99.97Hz 1024x768@75.03Hz 1024x768@70.07Hz 1024x768@60.00Hz 800x600@119.97Hz 800x600@99.66Hz 800x600@75.00Hz 800x600@72.19Hz 800x600@60.32Hz 800x600@56.25Hz 720x576@50.00Hz 720x480@59.94Hz 640x480@119.52Hz 640x480@99.77Hz 640x480@75.00Hz 640x480@72.81Hz 640x480@59.94Hz 640x480@59.93Hz
```

**Comment by Inevitable-Design** [comment]
**Not a solution**
I have the same issue, I couldn't yet find a solution but these are some workarounds you can try (if these help)
- Configure the displays to be placed diagonally
- Turn off one monitor during the game session (using hyprctl)

For me this behavior occurs in lot of games with exceptions. Minecraft java and lethal company doesn't have this issue for me.

**Comment by ranomier** [comment]
I have the same problem with valheim, which is a linux native game.

Even with window shortcuts it just doesn't move. I have animation off, so it shows on the correct monitor for a split second and forces itself back.

"alt tab" also doesn't work any more on that monitor.

I do not like that new change, i want every window to launch where my cursor is, no exception and the window has no power and that decision
